htmltools/_core.py

```diff
@@ -99,13 +109,23 @@ class MetadataNode:
 unnamed arguments to Tag functions like `div()`.
 """
 
-TagNode = Union["Tagifiable", "Tag", MetadataNode, "ReprHtml", str]
+# NOTE: If this type is updated, please update `is_tag_node()`
+TagNode = Union[
+    "Tagifiable",
+    # "Tag", # Tag is Tagifiable, do not include here
+    # "TagList", # TagList is Tagifiable, do not include here
```

Comment:
Add comment that says that `TagList` can't

```suggestion
    # "TagList" is Tagifiable, so it is included in practice, but in reality it should not be,
    #   because a TagList cannot contain a TagList.
```